sdk/src/wallet/operations/transaction/submit_transaction.rs

```diff
@@ -22,11 +20,10 @@ where
     ) -> crate::wallet::Result<BlockId> {
         log::debug!("[TRANSACTION] submit_signed_transaction");
 
-        #[cfg(feature = "events")]
-        self.emit(WalletEvent::TransactionProgress(TransactionProgressEvent::Broadcasting))
-            .await;
+        let block_id = self
+            .submit_basic_block(Some(Payload::from(payload)), issuer_id, true)
+            .await?;
 
-        self.submit_basic_block(Some(Payload::from(payload)), issuer_id, true)
-            .await
+        Ok(block_id)
```

Comment:
Can now just be `self.submit_basic_block(Some(Payload::from(payload)), issuer_id, true).await` again